### ISSUE TITLE
Specify when the model needs to be trained

### ISSUE THREAD

**Issue by perellonieto** [IRC-SPHERE/SklearnHyperStream#5]
There must be some method in the tools to specify when the model needs to be trained and when to stop the training. This could be integrated into the same model, but it could be as well some other stream of options that could be given alongside the data. E.g. <stream , metadata = (train: true, learning_rate: 0.1)>.

**Comment by tdiethe** [comment]
I'm assuming that the learning rate should be changing over time, otherwise it should just be a parameter.

A few options here:
- have a separate tool for training and prediction (maybe not ideal)
- augment the data stream so that it contains this information and gets unpacked by the tool
- provide a second stream that is aligned to the data stream, and contains packets with the metadata you suggest (e.g. as a dict)
